src/common/set.c

```diff
@@ -489,9 +494,17 @@ util_poolset_parse(const char *path, int fd, struct pool_set **setp)
 		return -1;
 	}
 
+	fd = dup(fd);
+	if (fd < 0) {
+		ERR("!dup");
+		return -1;
+	}
+
 	/* associate a stream with the file descriptor */
-	if ((fs = fdopen(dup(fd), "r")) == NULL) {
+	if ((fs = fdopen(fd, "r")) == NULL) {
```

Comment:
Is this change necessary? If `dup()` fails, it returns -1. So, this will be passed to `fdopen()`, and... I would expect it to fail as well. In such case, there is no need to close a duplicated file descriptor, as there is no one.
IMHO, there is no need to handle `dup()` failure separately. The only problem was missing `Free(set)`.